### PR TITLE
Make cluster and workspace views experimental

### PR DESCRIPTION
**Pull request by fjakobs**
## Changes
We still haven't decided if we want to keep these panels for good so we hide them behind a feature flag before GA.

## Tests

https://github.com/databricks/databricks-vscode/assets/40952/c97e1e91-648b-4732-9c54-0f62c569e69e

### CODE REVIEW

File: packages/databricks-vscode/src/extension.ts [hunk 1]
```diff
@@ -126,6 +126,33 @@ export async function activate(
         )
     );
 
+    // manage contexts for experimental features
+    function updateFeatureContexts() {
+        commands.executeCommand(
```
Comment (kartikgupta-db) [inline packages/databricks-vscode/src/extension.ts:131]: We have a `CustomWhenContext` object that allows manipulation of the context. 
https://github.com/databricks/databricks-vscode/pull/770/files#diff-6cb0770ff84b4a7174d5ef92c2d1e64dac98beb798d5bc4cbf4df2ce7b423f0dL53

Comment (fjakobs) [reply packages/databricks-vscode/src/extension.ts]: Perfect. Moved the code there.

File: packages/databricks-vscode/src/vscode-objs/CustomWhenContext.ts [hunk 2]
```diff
@@ -17,4 +17,28 @@ export const CustomWhenContext = {
...
+
+    updateShowClusterView() {
+        commands.executeCommand(
+            "setContext",
+            "databricks.feature.views.cluster",
+            (
+                workspace
+                    .getConfiguration("databricks")
+                    .get("experiments.optInto") as string[]
+            ).includes("views.cluster")
```
Comment (kartikgupta-db) [inline packages/databricks-vscode/src/vscode-objs/CustomWhenContext.ts]:
```suggestion
           workspaceConfigs.experimetalFeatureOverides.includes("views.cluster")
```

File: packages/databricks-vscode/src/vscode-objs/CustomWhenContext.ts [hunk 3]
```diff
@@ -17,4 +17,28 @@ export const CustomWhenContext = {
...
+
+    updateShowWorkspaceView() {
+        commands.executeCommand(
+            "setContext",
+            "databricks.feature.views.workspace",
+            (
+                workspace
+                    .getConfiguration("databricks")
+                    .get("experiments.optInto") as string[]
+            ).includes("views.workspace")
```
Comment (kartikgupta-db) [inline packages/databricks-vscode/src/vscode-objs/CustomWhenContext.ts]:
```suggestion     
workspaceConfigs.experimetalFeatureOverides.includes("views.workspace")
```

File: packages/databricks-vscode/src/vscode-objs/CustomWhenContext.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import {commands} from "vscode";
+import {commands, workspace} from "vscode";
```
Comment (kartikgupta-db) [inline packages/databricks-vscode/src/vscode-objs/CustomWhenContext.ts]:
```suggestion
import {commands, workspace} from "vscode";
import {workspaceConfigs} from "../WorkspaceConfigs";
```

**Review by kartikgupta-db — APPROVED** [review]

**Pull request merged**